src/SpineAnimation.ts

```diff
@@ -119,75 +141,157 @@ export class SpineAnimation extends Script {
+    super._updateRendererShaderData(context);
+    if (this._skeleton) {
+      this._spineGenerator.buildPrimitive(this._skeleton);
+    }
+  }
+
+  /**
+   * @internal
+   */
+  // @ts-ignore
+  protected override _render(context: any): void {
+    const primitive = this._primitive;
+    const subPrimitives = this._subPrimitives;
+    if (this._dirtyUpdateFlag & SpineAnimationUpdateFlags.VertexElementMacro) {
+      const shaderData = this.shaderData;
+      const vertexElements = primitive.vertexElements;
+
+      shaderData.disableMacro(SpineAnimation._uvMacro);
+      shaderData.disableMacro(SpineAnimation._enableVertexColorMacro);
+
+      for (let i = 0, n = vertexElements.length; i < n; i++) {
+        switch (vertexElements[i].attribute) {
+          case "TEXCOORD_0":
+            shaderData.enableMacro(SpineAnimation._uvMacro);
+            break;
+          case "COLOR_0":
+            shaderData.enableMacro(SpineAnimation._enableVertexColorMacro);
+            break;
+        }
+      }
+      this._dirtyUpdateFlag &= ~SpineAnimationUpdateFlags.VertexElementMacro;
+    }
+
+    const { _materials: materials, _engine: engine } = this;
+    // @ts-ignore
+    const renderElement = engine._renderElementPool.get();
+    // @ts-ignore
+    renderElement.set(this.priority, this._distanceForSort);
+    // @ts-ignore
+    const subRenderElementPool = engine._subRenderElementPool;
+    if (!subPrimitives) return;
+    for (let i = 0, n = subPrimitives.length; i < n; i++) {
+      let material = materials[i];
+      if (!material) {
+        continue;
+      }
+      if (material.destroyed || material.shader.destroyed) {
+        // @ts-ignore
+        material = this.engine._meshMagentaMaterial;
+      }
+
+      const subRenderElement = subRenderElementPool.get();
+      // @ts-ignore
+      subRenderElement.set(this, material, primitive, subPrimitives[i]);
+      renderElement.addSubRenderElement(subRenderElement);
     }
+    // @ts-ignore
+    context.camera._renderPipeline.pushRenderElement(context, renderElement);
   }
 
-  updateBounds() {
-    const meshRenderer = this.entity.getComponent(MeshRenderer);
-    const bounds = meshRenderer.bounds;
-    const offset = this._tempOffset;
-    const size = this._tempSize;
-    const temp = this._tempArray;
+  /**
+   * @internal
+   */
+  // @ts-ignore
+  protected override _updateBounds(worldBounds: BoundingBox): void {
+    const bounds = worldBounds;
+    const offset = SpineAnimation._tempOffset;
+    const size = SpineAnimation._tempSize;
+    const temp = SpineAnimation._tempArray;
     const zSpacing = this.setting?.zSpacing || 0.01;
     const skeleton = this._skeleton;
     skeleton.getBounds(offset, size, temp);
-    const drawOrder = skeleton.drawOrder;
-    bounds.min.set(offset.x, offset.y, 0);
-    bounds.max.set(
-      offset.x + size.x,
-      offset.y + size.y,
-      drawOrder.length * zSpacing
+    const { x, y } = offset;
```

Comment:
Because `buildPrimitive` is called every frame so the bounding box needs to be calculated every frame?